linode/manager
# refactor: [M3-7385]- MUI v5 Migration - `SRC > Features > Support`

## Description 📝

Performs general refactors on `SRC > Features > Support` to reduce tech debt, use MUI v5 features, and make Support tickets look _slightly_ less broken.

## Changes  🔄
- Rewrites some class components to functional components
- Makes the UI look a bit less broken
- Rewrite a few enzyme unit tests
- Runs the tss coded on 8 files

## Preview 📷

> [!note]
> Minimal UI changes are expected. Some styles may look different in attempts to make things look better.

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-11-07 at 12 46 44 PM](https://github.com/linode/manager/assets/115251059/b38bd0a5-9e29-4c6e-8a14-2e4464205e46) | ![Screenshot 2023-11-07 at 12 45 39 PM](https://github.com/linode/manager/assets/115251059/33b6bf11-d518-4394-aa32-6e4892d4fe3a) |

## How to test 🧪

> [!warning]
> I recommend testing against the dev environment so we don't create production support tickets

### Verification steps 
- Please check all features under http://localhost:3000/support/tickets
- Verify you can create a support ticket
- Verify you can reply to a support ticket
- Verify you can attach files to a support ticket

## As an Author I have considered 🤔
